### ISSUE TITLE
`base_score` is redundant for multinomial logistic

### ISSUE THREAD

**Issue by david-cortes** [dmlc/xgboost#11374]
After this PR: https://github.com/dmlc/xgboost/pull/10298

Objectives `multi:softprob` and `multi:softmax` (which correspond to multinomial logistic) calculate an intercept based on averaging values across all classes.

But this has no effect since adding a constant base score to all columns does not change the result after applying a softmax transformations - that is: softmax(a, b, c...) == softmax(a + const, b + const, c + const, ...).

It could skip the calculation altogether in such case.

**Comment by trivialfis** [comment]
I will move ahead and remove the binary model in this release: https://github.com/dmlc/xgboost/pull/11307

Afterward, we can have a vector for the base score.

**Comment by trivialfis** [comment]
> It could skip the calculation altogether in such case.

We still need the intercept as a base model for consistency. But you have a very good point. I can relax some constraints.

**Comment by david-cortes** [comment]
> > It could skip the calculation altogether in such case.
> 
> We still need the intercept as a base model for consistency. But you have a very good point. I can relax some constraints.

But you could leave it fixed at zero then. It wouldn't change the end result.

**Comment by trivialfis** [comment]
We are moving toward vector-valued intercept. https://github.com/dmlc/xgboost/pull/11656
